app/javascript/leaflet/utils.js

```diff
@@ -52,4 +52,27 @@ export const getTileJsonBounds = (doc) => {
       corner2 = latLng(bounds[3], bounds[2]);
     return latLngBounds(corner1, corner2);
   }
-};
\ No newline at end of file
+};
+
+export const appendLoadingMessage = () => {
+  const spinner = `<tbody class="attribute-table-body"><tr><td colspan="2">
+    <span id="attribute-table">
+    <div class="spinner-border" role="status"><span class="sr-only">Inspecting</span></div>
+    </span>
+    </td></tr></tbody>`;
+
+  document.querySelector(".attribute-table-body").innerHTML = spinner;
+};
+
+export const appendErrorMessage = () => {
+  document.querySelector(
+    ".attribute-table-body"
+  ).innerHTML = `<tbody class="attribute-table-body">
+    <tr><td colspan="2">Could not find that feature</td></tr></tbody>`;
+};
+
+// Looks for strings that could be URLs and wraps them in <a> tags
+export const linkify = (str) => {
+  const urlRegEx = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-]*)?\??(?:[\-\+=&;%@\.\w]*)#?(?:[\.\!\/\\\w]*))?)/g;
```

Comment:
This seems like a very curious regex.  Do we want it to support basic auth?  It seems odd it requires 3-9 letters (no websocket 'ws')  for the protocol. It has `www` embedded in it but doesn't allow other subdomains?

Reply:
Consider using:
```javascript
try {
  url = URL.new(str.toString()).toString()
  return `<a href='${url}'>${url}</a>`
 } catch(error) {
 return str.toString();
 }
 ```

Reply:
iirc this is old code and it's a vendorized version of a package that does this...maybe we should reevaluate or just use the package instead

Reply:
Oh...

Reply:
Yes this was ported over from the original code. We had trouble loading a linkify node module tbh with the new setup tbh so having our own method was easier to access. Happy to reevaluate.